deploy-service/common/src/main/java/com/pinterest/deployservice/common/DBConfigReader.java

```diff
@@ -81,10 +83,22 @@ public Integer getPort(String replicaSetName) {
     }
 
     public String getUsername(String role) {
-        return JsonPath.read(credJson, String.format("$.%s.users[?(@.enabled == true)][0].username", role));
+        JSONArray jsonArray = JsonPath.read(credJson, String.format("$.%s.users[?(@.enabled == true)]", role));
+        if(jsonArray.size()>0){
+            Map hashMap =  (Map) jsonArray.get(0);
```

Comment:
Should we not initialize a new `HashMap` with the `new` keyword? What's the structure of `jsonArray.get(0)`?